decoder/body_extensions_test.go

```diff
@@ -460,6 +460,452 @@ variable "test" {
 				},
 			}),
 		},
+		{
+			"for_each attribute expression completion",
+			&schema.BodySchema{
+				Blocks: map[string]*schema.BlockSchema{
+					"resource": {
+						Labels: []*schema.LabelSchema{
+							{
+								Name: "type",
+							},
+							{
+								Name: "name",
+							},
+						},
+						Body: &schema.BodySchema{
+							Extensions: &schema.BodyExtensions{
+								ForEach: true,
+							},
+						},
+					},
+				},
+			},
+			reference.Targets{
+				{
+					Addr: lang.Address{
+						lang.RootStep{Name: "var"},
+						lang.AttrStep{Name: "test"},
+					},
+					ScopeId: lang.ScopeId("variable"),
+					Type:    cty.Map(cty.String),
+					RangePtr: &hcl.Range{
+						Filename: "test.tf",
+						Start: hcl.Pos{
+							Line:   4,
+							Column: 1,
+							Byte:   48,
+						},
+						End: hcl.Pos{
+							Line:   6,
+							Column: 2,
+							Byte:   87,
+						},
+					},
+				},
+			},
+			`resource "aws_instance" "foo" {
+  for_each = 
+}
+variable "test" {
+	type = map(string)
+}
+`,
+			hcl.Pos{
+				Line:   2,
+				Column: 14,
+				Byte:   45,
+			},
+			lang.CompleteCandidates([]lang.Candidate{
+				{
+					Label:  "each.key",
+					Detail: "string",
+					Description: lang.MarkupContent{
+						Value: "The map key (or set member) corresponding to this instance",
+						Kind:  lang.MarkdownKind,
+					},
+					TextEdit: lang.TextEdit{
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start: hcl.Pos{
+								Line:   2,
+								Column: 14,
+								Byte:   45,
+							},
+							End: hcl.Pos{
+								Line:   2,
+								Column: 14,
+								Byte:   45,
+							},
+						},
+						NewText: "each.key",
+						Snippet: "each.key",
+					},
+					Kind: lang.TraversalCandidateKind,
+				},
+				{
+					Label:  "each.value",
+					Detail: "string",
+					Description: lang.MarkupContent{
+						Value: "The map value corresponding to this instance. (If a set was provided, this is the same as `each.key`.)",
+						Kind:  lang.MarkdownKind,
+					},
+					TextEdit: lang.TextEdit{
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start: hcl.Pos{
+								Line:   2,
+								Column: 14,
+								Byte:   45,
+							},
+							End: hcl.Pos{
+								Line:   2,
+								Column: 14,
+								Byte:   45,
+							},
+						},
+						NewText: "each.value",
+						Snippet: "each.value",
+					},
+					Kind: lang.TraversalCandidateKind,
+				},
```

Comment:
```suggestion
```
As mentioned in my other comment in-line, these two references shouldn't be available in the context of `for_each` directly, that would be kinda like a reference to itself.